### PR TITLE
Added hexify

### PR DESCRIPTION
**Pull request by ZuluPro**
### What does this changes

In the same spirit than lexify, I added hexify, to format strings with hexadecimal numbers.

I could be useful in many case, especially to develop providers.

### CODE REVIEW

File: faker/providers/__init__.py [hunk 1]
```diff
@@ -213,3 +213,17 @@ def bothify(self, text='## ??', letters=string.ascii_letters):
         :returns: string with all numerical and letter placeholders filled in
         """
         return self.lexify(self.numerify(text), letters=letters)
+
+    def hexify(self, text='????', upper=False):
```
Comment (fcurella) [inline faker/providers/__init__.py]: I would try to come up with a different character than `?` or `#`, maybe `x`?

Comment (ZuluPro) [reply faker/providers/__init__.py]: No, `x` is too ASCII, more `$` ?

Comment (fcurella) [reply faker/providers/__init__.py]: how about `×` (unicode 'multiplication sign')?

Comment (ZuluPro) [reply faker/providers/__init__.py]: How do you do it ? 😄

Comment (fcurella) [reply faker/providers/__init__.py]: good point, i think it's too inconvenient. I'm worried `$` is too close to the idea of 'currency'. Maybe we could use `*`, `^` or `&`

Comment (ZuluPro) [reply faker/providers/__init__.py]: What about `~` ?
Easy to do for everyone, without being used all the time.

Comment (ZuluPro) [reply faker/providers/__init__.py]: In fact, forget my last comment, let's do `^` !

**Comment by ZuluPro** [comment]
@fcurella Changed !

**Comment by fcurella** [comment]
Thank you!

**Pull request merged**